### ISSUE TITLE
playview.io Anti Adblock

### ISSUE THREAD

**Issue by xDarkARG** [reek/anti-adblock-killer#3738]
https://playview.io/

**Comment by uBlock-user** [comment]
@jspenguin2017 destroys the page to all black

**Comment by uBlock-user** [comment]
nooping the XHR doesn't work !

**Comment by jspenguin2017** [comment]
loopback would work but, meh, I'll just white list, not worth using loopback on that.

**Comment by uBlock-user** [comment]
```javascript
$(document).ready(function(){function adBlockDetected(){$('#allcontent').remove();$("#showaviso").show();}
function adBlockNotDetected(){$("#showaviso").remove();}
if($('#showaviso').length==1){$.ajax('themes/pv/js/showads.js').fail(function(d){if(d.status===0||d.statusText=='error'){adBlockDetected();$("body").html('<div id="showaviso" class="container"><div class="row" style="padding:50px;"><div class="jumbotron text-center"><img src="media/adblock.png" class="imb-responsive center-block"/><br>Hemos detectado que tienes AdBlock y estÃ¡ habilitado para nuestro sitio lamentablemente nos mantenemos de la publicidad que hay aquÃ­ y si evitas la publicidad no generamos ganancias para poder seguir online, por favor desactiva AdBlock para nuestro sitio y actualiza nuevamente para visualizar el contenido</div></div></div>');}});if(typeof fuckAdBlock==='undefined'){adBlockDetected();}else{fuckAdBlock.setOption({debug:false});fuckAdBlock.onDetected(adBlockDetected).onNotDetected(adBlockNotDetected);}}
function checkAgain(){$('#fuck-adb-enabled').hide();$('#fuck-adb-not-enabled').hide();setTimeout(function(){if(typeof fuckAdBlock==='undefined'){adBlockDetected();}else{fuckAdBlock.onDetected(adBlockDetected).onNotDetected(adBlockNotDetected);fuckAdBlock.check();}},300);}
```

Can something else be done ?

**Comment by jspenguin2017** [comment]
It's in a closure. There are a few ways to get around it, but most are pretty expensive. 

Patching through this closure is left as an exercise to the reader. :trollface:

**Comment by uBlock-user** [comment]
Yeah looks like that, screw it, whitelist it is!

**Comment by lain566** [comment]
When I try to play a video, I can't click on the player
`https://playview.io/the-gifted-temp-1#Episode-4`

**Comment by uBlock-user** [comment]
@okiehsch can you tell why redirection filter doesn't work here ?

**Comment by jspenguin2017** [comment]
It choked on CORS, like always. 
`a.loopbackXHR` would work, but it's too expensive for what we are trying to achieve.

**Comment by jspenguin2017** [comment]
Will need content script property patching before this gets fixed: https://bugs.chromium.org/p/chromium/issues/detail?id=749236

**Comment by uBlock-user** [comment]
Does this website have a CSP policy ? I didn't see any in the console!

**Comment by jspenguin2017** [comment]
Huh, that's weird... 
Lemme read jQuery doc real quick.

**Comment by uBlock-user** [comment]
Okay I'm seeing this once I apply the redirect filter

`Failed to load data:text/plain;base64,: Request header field X-Requested-With is not allowed by Access-Control-Allow-Headers in preflight response.`

**Comment by jspenguin2017** [comment]
Yea, that explains it, jQuery will insert `X-Requested-With` header. 
I was reading jQuery source, `$.ajax` is bloated beyond belief. 436 lines uncompressed...

**Comment by uBlock-user** [comment]
True, it's being inserted from `jquery.min.js` which is being served from `ajax.googleapis.com`

**Comment by uBlock-user** [comment]
So to summarise, https://bugs.chromium.org/p/chromium/issues/detail?id=749236 is affecting the redirect filter ?

**Comment by jspenguin2017** [comment]
That's a good question, it's not really CSP but still extension hitting on security features. 
If we have a good patch it'll probably be accepted, except that not many people has a skill and time to make a patch.

**Comment by uBlock-user** [comment]
Wow, so we're screwed until someone comes up with a patch. I wonder if FF 57 is also affected by this.

**Comment by jspenguin2017** [comment]
I wouldn't say screwed, as `a.loopbackXHR` will work in case we need it. 

Patching Chromium is no easy task, and that's beyond my current skill level. For it to be "properly" fixed, we'll probably need to wait for someone else.

**Comment by uBlock-user** [comment]
Patching Chromium isn't your task in any way either, Google has a battalion of devs, they're the ones that're suppose to handle these shortcomings in the first place.

**Comment by jspenguin2017** [comment]
They have better things to do than helping us blocking their ads. They did say that they'll accept a patch if it's good, but chances that they'll make the patch themselves is slim. 
It is some shortcoming with extension permissions, well, it's shorter on Firefox so...

**Comment by uBlock-user** [comment]
They created this issue in the first place, so it is their job to fix things which they themselves bugged it from the beggining. The issue tracker doesn't have an "owner" and as long as it doesn't have an "owner", the patch will never land.
